core/unit_test/TestViewAPI.hpp

```diff
@@ -837,18 +837,15 @@ struct TestViewMirror {
                                                     view_const_cast(v));
   }
 
-  template <class MemoryTraits, class Space>
+  template <class View>
   struct CopyUnInit {
-    using mirror_view_type = typename Kokkos::Impl::MirrorViewType<
-        Space, double *, Layout, Kokkos::HostSpace, MemoryTraits>::view_type;
-
-    mirror_view_type a_d;
+    View a_d;
 
     KOKKOS_INLINE_FUNCTION
-    CopyUnInit(mirror_view_type &a_d_) : a_d(a_d_) {}
+    explicit CopyUnInit(View &a_d_) : a_d(a_d_) {}
```

Comment:
```suggestion
    explicit CopyUnInit(View const& a_d_) : a_d(a_d_) {}
```

That should allow you to use CTAD later `CopyUnInit(a_d)`.

Reply:
It didn't help with Intel classics.